pallets/dynamic-evm-fee/src/lib.rs

```diff
@@ -138,19 +159,27 @@ pub mod pallet {
 						.saturating_mul(3),
 				);
 
-				let Some(eth_hdx_price) = T::NativePriceOracle::price(T::WethAssetId::get()) else {
+				let evm_asset = EvmAsset::<T>::get().unwrap_or(T::WethAssetId::get());
+
+				let Some((eth_per_hdx, eth_per_hdx_reference)) = T::EvmAssetPrices::get(&evm_asset) else {
 					log::warn!(target: "runtime::dynamic-evm-fee", "Could not get ETH-HDX price from oracle");
 					return;
 				};
-				let Some(eth_hdx_price) = FixedU128::checked_from_rational(eth_hdx_price.n, eth_hdx_price.d) else {
-					log::warn!(target: "runtime::dynamic-evm-fee", "Could not get rational of eth-hdx price, n: {}, d: {}", eth_hdx_price.n, eth_hdx_price.d);
+				let Some(eth_per_hdx) = FixedU128::checked_from_rational(eth_per_hdx.n, eth_per_hdx.d) else {
+					log::warn!(target: "runtime::dynamic-evm-fee", "Could not get rational of eth-hdx price, n: {}, d: {}", eth_per_hdx.n, eth_per_hdx.d);
 					return;
 				};
 
+				let Some(eth_per_hdx_reference) =
+					FixedU128::checked_from_rational(eth_per_hdx_reference.n, eth_per_hdx_reference.d)
+				else {
+					log::warn!(target: "runtime::dynamic-evm-fee", "Could not get rational of eth-hdx reference price, n: {}, d: {}", eth_per_hdx_reference.n, eth_per_hdx_reference.d);
+					return;
+				};
 				let Some(price_diff) =
```

Comment:
nitpick:

it is not really a price difference when you divide one price by another price.

it is a price ratio, or relative price.